bentoml/_internal/frameworks/fastai.py

```diff
@@ -0,0 +1,285 @@
+import typing as t
+import logging
+from typing import TYPE_CHECKING
+from contextlib import ExitStack
+
+import cloudpickle
+
+import bentoml
+from bentoml import Tag
+
+from ..types import LazyType
+from ..utils.pkg import get_pkg_version
+from ...exceptions import NotFound
+from ...exceptions import InvalidArgument
+from ...exceptions import BentoMLException
+from ...exceptions import MissingDependencyException
+from ..models.model import ModelContext
+from .common.pytorch import PyTorchTensorContainer  # type: ignore
+
+MODULE_NAME = "bentoml.fastai"
+MODEL_FILENAME = "saved_model.pkl"
+API_VERSION = "v1"
+
+logger = logging.getLogger(__name__)
+
+
+if TYPE_CHECKING:
+    from .. import external_typing as ext
+    from ...types import ModelSignature
+    from ..models.model import ModelSignaturesType
+
+try:
+    import torch
+    import torch.nn as nn
+except ImportError:  # pragma: no cover
+    raise MissingDependencyException(
+        "fastai requires `torch` as a dependency. Please follow PyTorch instruction at https://pytorch.org/ in order to use `fastai`."
+    )
+try:
+    from fastai import __version__ as FASTAI_VERSION  # type: ignore
+    from fastai.learner import Learner
+    from fastai.learner import load_learner  # type: ignore
+except ImportError:  # pragma: no cover
+    raise MissingDependencyException(
+        "fastai is required in order to use module `bentoml.fastai`, install fastai with `pip install fastai`. For more information, refers to https://docs.fast.ai/#Installing."
+    )
+
+try:
+    import fastai.basics  # type: ignore
+except ImportError:  # pragma: no cover
+    raise MissingDependencyException(
+        "BentoML will only support fastai v2 onwards. Please uninstall fastai v1 and try again. Please visit https://docs.fast.ai/#Installing for more information."
+    )
+
+
+__all__ = ["load_model", "save_model", "get_runnable", "get"]
+
+
+def get(tag_like: str | Tag) -> bentoml.Model:
+    """
+    Get the BentoML model with the given tag.
+
+    Args:
+        tag_like (``str`` ``|`` :obj:`~bentoml.Tag`):
+            The tag of the model to retrieve from the model store.
+    Returns:
+        :obj:`~bentoml.Model`: A BentoML :obj:`~bentoml.Model` with the matching tag.
+
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+        # target model must be from the BentoML model store
+        model = bentoml.fastai.get("fai_learner")
+    """
+    model = bentoml.models.get(tag_like)
+    if model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {model.tag} was saved with module {model.info.module}, failed loading with {MODULE_NAME}."
+        )
+    return model
+
+
+def load_model(
+    bento_model: str | Tag | bentoml.Model,
+    cpu: bool = True,
+    pickle_module: types.ModuleType = cloudpickle,
+) -> Learner:
+    """
+    Load the :code:`fastai.learner.Learner` model instance with the given tag from the local BentoML model store.
```

Comment:
We should probably standardize on either using double backtick or `:code:`.

Reply:
i prefer to have `:code:` block

Reply:
No real preference, but generally `:code:` is more intrusive when just reading the doc comment is my 2 cents.